### ISSUE TITLE
Ram editor should search all readable & writeable memory region

### ISSUE THREAD

**Issue by zouquan741** [WerWolv/EdiZon#54]
**Is your feature request related to a problem? Please describe.**
I found that ram editor can't find out the address of gold in some game, like Tales of Vesperia.

**Describe the solution you'd like**
I did it by using Noexes, and notice that the address of gold located in code_XXXX region, not heap.
Maybe these value is stored in static var?
I hope EdiZon can support it.

**Comment by WerWolv** [comment]
That will be done by a separate pointer search mode.
When just using EdiZon as a cheat engine you really don't want to search other regions except code as there will be a lot of useless values then

**Comment by zouquan741** [comment]
Maybe add a search region option like what Noexes did?
Also I don't understand why pointer search mode can solve this problem.

**Comment by WerWolv** [comment]
Pointer search mode would search both the code and the heap region for pointers. You don't want to search the code region when just looking for values in RAM

**Comment by zouquan741** [comment]
But there is some value store in code region, not their pointer. 
Tales of Vesperia for example, most of value I want to modify do located in code region, like gold, HP, TP...
I have tested, none of this value can found by EdiZon.
Though I can search value under pointer search mode, but 64 bit width value only.

**Comment by WerWolv** [comment]
Wait, there are actually games that write their values to the main nso? How weird...

**Comment by zouquan741** [comment]
I think it is quite common in japanese game. 
I have found two game who did these, "Atelier Lydie & Suelle" and "Tales of Vesperia".
I will post some screenshot of Noexes later.

**Comment by zouquan741** [comment]
![捕获](https://user-images.githubusercontent.com/13255981/54086759-5d82a600-4387-11e9-941c-2a06db467cf4.JPG)
![捕获2](https://user-images.githubusercontent.com/13255981/54086761-5f4c6980-4387-11e9-89dc-e7c254bb7bec.JPG)
![Inked2019031022411200-64902EF0DF1ABA05E4076CB0BBFCD9A5_LI](https://user-images.githubusercontent.com/13255981/54086763-61aec380-4387-11e9-94b8-1b758e1643fe.jpg)

**Comment by zouquan741** [comment]
Here is my modified version, now it works perfectly!
https://github.com/zouquan741/EdiZon/commit/18f2378523aedc5db6b4993c64b4ded35e28139f

**Comment by AnalogMan151** [comment]
I agree with this. For example right now I needed to update an ASM code for a game that had an update. This was made easy by searching for the u32 instruction from the previous version of the game and finding the new memory location for that same instruction. Something I would not have been able to do with the pointer search.

**Comment by satelliteseeker** [comment]
Based on @zouquan741 's analysis, I would prefer adding `MemType_CodeMutable` as a searchable memory region by default.

As for other regions, they could be searched in a new "extended mode" (toggled in config) as while they may be useful in game modding, too many garbage result will be produced for regular users.

**Comment by WerWolv** [comment]
@satelliteseeker 
There will be 4 modes. Heap search, NSO search, Heap + NSO search and a search that covers the entire memory regardless of memory region
